client/src/cbltest/api/syncgateway.py

```diff
@@ -1340,6 +1342,352 @@ async def fetch_log_file(
+            ssh.close()
+
+    async def wait_for_database_to_be_offline(self, db_name: str) -> None:
+        """
+        Waits for a database to be offline
+
+        :param db_name: Database name to wait for
+        """
+        max_attempts = 6
+        wait_time = 5
+        for _ in range(max_attempts):
+            try:
+                resp = await self._send_request("get", f"/{db_name}")
+                if resp.code == 403:
+                    return
+            except Exception as e:
+                if isinstance(e, CblSyncGatewayBadResponseError) and e.code == 403:
+                    return
+                await asyncio.sleep(wait_time)
+        raise Exception(
+            f"Database {db_name} did not go offline after {max_attempts * wait_time} seconds."
+        )
+
+    async def scan_rest_endpoints(
+        self,
+        db_name: str,
+        expected_online: bool,
+        num_docs: int = 10,
+    ) -> tuple[int, int]:
+        """
+        Scans multiple REST endpoints to verify database online/offline state.
+
+        :param db_name: Database name to test
+        :param expected_online: True if DB should be online, False if offline
+        :param num_docs: Number of docs (for testing doc operations)
+        :return: Tuple of (endpoints_tested, errors_503)
+        """
+        endpoints_tested = 0
+        errors_403 = 0
+
+        # List of endpoints to test
+        test_operations = [
+            (
+                "GET /{db}/_all_docs",
+                lambda: self.get_all_documents(db_name, "_default", "_default"),
+            ),
+            (
+                "GET /{db}/_changes",
+                lambda: self.get_changes(db_name, "_default", "_default"),
+            ),
+            (
+                "GET /{db}/{doc}",
+                lambda: self.get_document(
+                    db_name, f"{db_name}_doc_0", "_default", "_default"
+                ),
+            ),
+            (
+                "POST /{db}/_bulk_docs",
+                lambda: self.update_documents(
+                    db_name,
+                    [DocumentUpdateEntry("test_doc", None, {"foo": "bar"})],
+                    "_default",
+                    "_default",
+                ),
+            ),
+        ]
+        for endpoint_name, test_func in test_operations:
+            try:
+                await test_func()
+                endpoints_tested += 1
+            except (CblSyncGatewayBadResponseError, JSONDecodeError) as e:
+                endpoints_tested += 1
+                if isinstance(e, CblSyncGatewayBadResponseError) and e.code == 403:
+                    errors_403 += 1
+                elif isinstance(e, JSONDecodeError):
+                    errors_403 += 1
+                else:
+                    raise e
+            except Exception as e:
+                endpoints_tested += 1
+                raise e
+        return (endpoints_tested, errors_403)
+
+
+def verify_sensitive_data_is_hashed(
```

Comment:
Please split methods like this out into another class.  The Sync Gateway API should remain clean and familiar with Sync Gateway operations.  These are test logic specific.